src/TS/animate.ts

```diff
@@ -1,12 +1,9 @@
 function animate() {
     animationID = requestAnimationFrame(animate);
     enemies = enemies.filter((value, index) => {
-        let tr = !(value.id in enemiesToRemove)
-        if (tr) { enemiesToRemove.splice(index, 1) }
```

Comment:
This would cause issues, as it is the index of the enemies, not the index of the enemiesToRemove list